core/web/migrate_controller.go

```diff
@@ -91,11 +91,27 @@ func (mc *MigrateController) Migrate(c *gin.Context) {
 	jsonAPIResponse(c, webpresenters.NewJobResource(jb), jb.Type.String())
 }
 
-// Does not support mixed initiator types.
+// Does not support mixed initiator types, except for web/cron.
 func (mc *MigrateController) MigrateJobSpec(c *config.Config, js models.JobSpec) (job.Job, error) {
 	var jb job.Job
-	if len(js.Initiators) == 0 {
+	switch len(js.Initiators) {
+	case 0:
 		return jb, errors.New("initiator required to migrate job")
+	case 2:
+		// Special case for cron/web pairs: drop web
+		a, b := js.Initiators[0].Type, js.Initiators[1].Type
+		if a == models.InitiatorCron && b == models.InitiatorWeb {
+			js.Initiators = js.Initiators[:1]
+			break
+		} else if a == models.InitiatorWeb && b == models.InitiatorCron {
+			js.Initiators = js.Initiators[1:]
+			break
+		}
+		fallthrough // unsupported pair
+	default:
+		return jb, fmt.Errorf("mixed initiator types unsupported: %d initiators", len(js.Initiators))
```

Comment:
Doesn't matter much since its throwaway code but I wouldn't be consistent if I didn't mention using `errors.Errorf` here instead for stacktraces